connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -608,6 +608,20 @@ private void readTaskStatus(String key, byte[] value) {
         synchronized (this) {
             log.trace("Received task {} status update {}", id, status);
             CacheEntry<TaskStatus> entry = getOrAdd(id);
+
+            // During frequent rebalances, there could be a race condition because of which
+            // an UNASSIGNED state of a prior generation can be sent by a worker despite a
+            // RUNNING status in a newer generation by another worker because the first worker
+            // couldn't read the newer RUNNING status. This can lead to an inaccurate status
+            // representation even though the task might be actually running.
+            if (status.state() == ConnectorStatus.State.UNASSIGNED
```

Comment:
May I ask should it be TaskStatus.State instead of ConnectorStatus.State in this method?

Reply:
Thats'a good catch @Lucent-Wong . Since both extend from `AbstractStatus` I didn't realise I was using `ConnectorStatus`. I changed it.